packages/core/src/component.ts

```diff
@@ -3,6 +3,31 @@
 // This file is licensed under the MIT License.
 // License text available at https://opensource.org/licenses/MIT
 
+import {Constructor, Provider} from '@loopback/context';
+import {Application} from '.';
+
+// tslint:disable:no-any
+
 export interface Component {
+  controllers?: Constructor<any>[];
+  repositories?: Constructor<any>[];
+  providers?: {
+    [key: string]: Constructor<Provider<any>>;
+  };
+}
+
+export function mountComponent(app: Application, component: Component) {
```

Comment:
I am intentionally keeping the implementation of component mounting in the same file where `Component` interface is defined. That way when we decide to add a new component feature (like repositories) later, then most of the changes can stay in this one file.